src/polybench/POLYBENCH_FDTD_2D.cpp

```diff
@@ -53,10 +53,27 @@ POLYBENCH_FDTD_2D::POLYBENCH_FDTD_2D(const RunParams& params)
       break;
   }
 
-  setDefaultSize( std::max( (m_nx-1)*m_ny, m_nx*(m_ny-1) ) );
+#if 0 // we want this...
+
+  Index_type nx_default = 800;
+  Index_type ny_default = 1260;
+
+  setDefaultProblemSize( std::max( (nx_default-1) * ny_default, 
+                                    nx_default * (ny_default-1) ) );
+  setDefaultReps(8);
+
+  m_nx = getTargetProblemSize() / ny_default;
```

Comment:
guard against 0

Reply:
Will fix.

Reply:
Add 1 after the divide?

Reply:
Maybe use max? std::max(getTargetProblemSize() / ny_default, 1)

Reply:
I suppose we could let the actual problem size be 0 if the target problem size is 0. 
```suggestion
  m_nx = (getTargetProblemSize()+ny_default-1) / ny_default;
```

Reply:
Fine with me.

Reply:
Actually, where it may break is if this leads to set number of threads or blocks to 0. I think that would cause an error. Maybe not here, but in some problem.

Reply:
I think I'm going to try another approach to scaling some of the polybench kernels.  I'll push some changes.

Reply:
You'd get 0 size for all of the normal kernels that just use the target size as well when target size is 0. I'm fine with not worrying about what happens when target size goes to 0 as that's not an interesting case anyway.